packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -481,13 +482,29 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             });
         }
 
+        let storage = context.storage;
+        if (context.baseSnapshot) {
+            // This will patch snapshot in place!
+            // If storage is provided, it will wrap storage with BlobAggregatorStorage that can
+            // pack & unpack aggregated blobs.
+            // Note that if storage is provided later by loader layer, we will wrap storage in this.storage getter.
+            // BlobAggregatorStorage is smart enough for double-wrapping to be no-op
+            if (context.storage) {
+                const aggrStorage = BlobAggregatorStorage.wrap(context.storage);
```

Comment:
We should add a way for the storage to declare whether we want this behavior or not.  This behavior might not be desirable for r11s.

Reply:
i think you always need it. it doesn't write anything compressed if minBlobSize in undefined, but regardless can read the compressed format

---
In reply to: [569728765](https://github.com/microsoft/FluidFramework/pull/5032#discussion_r569728765) [](ancestors = 569728765)